Doc/move database setup instructions in install md (#8987)
matrix-org/synapse: INSTALL.md

  - [Choosing your server name](#choosing-your-server-name)
  - [Installing Synapse](#installing-synapse)
    - [Installing from source](#installing-from-source)
      - [Platform-Specific Instructions](#platform-specific-instructions)
        - [Debian/Ubuntu/Raspbian](#debianubunturaspbian)
        - [ArchLinux](#archlinux)
        - [CentOS/Fedora](#centosfedora)
        - [macOS](#macos)
        - [OpenSUSE](#opensuse)
        - [OpenBSD](#openbsd)
        - [Windows](#windows)
    - [Prebuilt packages](#prebuilt-packages)
      - [Docker images and Ansible playbooks](#docker-images-and-ansible-playbooks)
      - [Debian/Ubuntu](#debianubuntu)
        - [Matrix.org packages](#matrixorg-packages)
        - [Downstream Debian packages](#downstream-debian-packages)
        - [Downstream Ubuntu packages](#downstream-ubuntu-packages)
      - [Fedora](#fedora)
      - [OpenSUSE](#opensuse-1)
      - [SUSE Linux Enterprise Server](#suse-linux-enterprise-server)
      - [ArchLinux](#archlinux-1)
      - [Void Linux](#void-linux)
      - [FreeBSD](#freebsd)
      - [OpenBSD](#openbsd-1)
      - [NixOS](#nixos)
  - [Setting up Synapse](#setting-up-synapse)

>     - [Switching the database engine](#switching-the-database-engine)

Reviewer: ```suggestion
    - [Using PostgreSQL](#using-postgresql)
```

Oops, missed a change here.

Installing Synapse:

```sh
doas pkg_add synapse
```

#### NixOS

Robin Lambertz has packaged Synapse for NixOS at:
<https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/misc/matrix-synapse.nix>

## Setting up Synapse

Once you have installed synapse as above, you will need to configure it.

> ### Switching the database engine

Reviewer: ```suggestion
### Using PostgreSQL
```

> By default Synapse gets setup using [SQLite](https://sqlite.org/) and in doing so trades performance for convenience.

Reviewer: ```suggestion
By default Synapse uses [SQLite](https://sqlite.org/) and in doing so trades performance for convenience.
```

SQLite is only recommended in Synapse for testing purposes or for servers with

> light workloads.

Reviewer: ```suggestion
very light workloads.
```

Just for that extra bit of convincing.

Almost all installations should opt to use [PostgreSQL](https://www.postgresql.org). Advantages include:

- significant performance improvements due to the superior threading and
  caching model, smarter query optimiser
- allowing the DB to be run on separate hardware

> For information on how to install and use PostgreSQL, please see

Reviewer: ```suggestion
For information on how to install and use PostgreSQL in Synapse, please see
```

Once you have installed synapse as above, you will need to configure it.

### Using PostgreSQL

By default Synapse uses [SQLite](https://sqlite.org/) and in doing so trades performance for convenience.
SQLite is only recommended in Synapse for testing purposes or for servers with
very light workloads.

Almost all installations should opt to use [PostgreSQL](https://www.postgresql.org). Advantages include:

- significant performance improvements due to the superior threading and
  caching model, smarter query optimiser
- allowing the DB to be run on separate hardware

> For information on how to install and use PostgreSQL in Synapse, please see

Author: Aah! Even better

connect to your server. This is the same URL you put for the `m.homeserver`
`base_url` above.

> ```nginx

Reviewer: ```suggestion
```yaml
```

This is supposed to happen in `homeserver.yaml`, per the paragraph above.
